### PR TITLE
[SYCL][NATIVECPU] Added remaining UR interface functions, PI exports and ABI Check

### PR DESCRIPTION
**Pull request by uwedolinsky**
This PR adds the remaining (not fully implemented) UR interface functions to the NativeCPU/UR adapter. 
The NativeCPU PI plugin now also exports all PI interface functions and an ABI check was added.

### CODE REVIEW

File: sycl/plugins/unified_runtime/ur/adapters/native_cpu/command_buffer.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+
+#include "common.hpp"
+
+/// Stub implementations of UR experimental feature command-buffers
+/// Taken almost unchanged from another adapter. Perhaps going forward
+/// these stubs could be defined in core UR as the default which would
```
Comment (jandres742) [inline sycl/plugins/unified_runtime/ur/adapters/native_cpu/command_buffer.cpp:13]: possibly, but that would imply having C++ headers for the definition of the UR interfaces, to allow adapters to have overrides. Each adapter needs to implement the stubs with their internal logic. But maybe in the future, some of the logic for command buffers could have a shared part, common to all adapters, and then other parts that are adapter-specific.

**Review by jandres742 — APPROVED** [review]

**Comment by jandres742** [comment]
+1 on common code.

**Review by PietroGhg — APPROVED** [review]

**Review by againull — APPROVED** [review]

**Comment by uwedolinsky** [comment]
Hello @intel/bindless-images-reviewers , please could you review this PR? Thank you!

**Review by Seanst98 — APPROVED** [review]

**Comment by uwedolinsky** [comment]
Hello @intel/llvm-gatekeepers , we think this PR is ready for merge. Thank you!

**Pull request merged**